File: Codes from the AI-assistants/[CLBG] mandelbrot/Basic Prompt - TBR Mode/Mandelbrot-deepseek-basic-tbr.py
import sys
import array

def mandelbrot(N, filename):
    # Precompute constants
    x_min, x_max = -1.5, 0.5
    y_min, y_max = -1.0, 1.0
    x_step = (x_max - x_min) / (N - 1)
    y_step = (y_max - y_min) / (N - 1)
    
    max_iter = 1000
    
    with open(filename, 'wb') as f:
        # Write PBM header
        f.write(f"P4\n{N} {N}\n".encode())
        
        # Process image row by row
        for y in range(N):
            c_imag = y_min + y * y_step
            row_bytes = array.array('B')
            byte = 0
            bit_count = 0
            
            for x in range(N):
                c_real = x_min + x * x_step
                z_real, z_imag = 0.0, 0.0
                iter_count = 0
                
                # Quadratic recurrence
                while iter_count < max_iter:
                    z_real_sq = z_real * z_real
                    z_imag_sq = z_imag * z_imag
                    
                    if z_real_sq + z_imag_sq > 4.0:
                        break
                    
                    # z_{n+1} = z_n^2 + c
                    new_real = z_real_sq - z_imag_sq + c_real
                    new_imag = 2 * z_real * z_imag + c_imag
                    z_real, z_imag = new_real, new_imag
                    iter_count += 1
                
                # Pack bits into bytes
                byte <<= 1
                if iter_count == max_iter:
                    byte |= 1  # Inside set (black)
                
                bit_count += 1
                if bit_count == 8:
                    row_bytes.append(byte)
                    byte = 0
                    bit_count = 0
            
            # Handle partial byte at end of row
            if bit_count > 0:
                byte <<= (8 - bit_count)
                row_bytes.append(byte)
            
            row_bytes.tofile(f)

if __name__ == "__main__":
    N = 16000
    if len(sys.argv) > 1:
        N = int(sys.argv[1])
    
    mandelbrot(N, "mandelbrot.pbm")

File: Codes from the AI-assistants/[CLBG] mandelbrot/Basic Prompt - TBR Mode/Mandelbrot-deepseek-basic-tbr.cpp
#include <fstream>
#include <complex>
#include <vector>

void mandelbrot(int N, const std::string& filename) {
    std::ofstream file(filename, std::ios::binary);
    if (!file) return;
    
    // PBM header
    file << "P4\n" << N << " " << N << "\n";
    
    const double x_min = -1.5, x_max = 0.5;
    const double y_min = -1.0, y_max = 1.0;
    const double x_step = (x_max - x_min) / (N - 1);
    const double y_step = (y_max - y_min) / (N - 1);
    
    const int max_iter = 1000;
    
    for (int y = 0; y < N; ++y) {
        const double c_imag = y_min + y * y_step;
        std::vector<unsigned char> row_bytes((N + 7) / 8, 0);
        
        for (int x = 0; x < N; ++x) {
            const double c_real = x_min + x * x_step;
            std::complex<double> z(0, 0);
            std::complex<double> c(c_real, c_imag);
            int iter = 0;
            
            // Quadratic recurrence
            while (iter < max_iter) {
                z = z * z + c;
                if (std::norm(z) > 4.0) break;
                ++iter;
            }
            
            if (iter == max_iter) {
                // Set bit for point in Mandelbrot set
                const int byte_index = x / 8;
                const int bit_index = 7 - (x % 8);
                row_bytes[byte_index] |= (1 << bit_index);
            }
        }
        
        file.write(reinterpret_cast<const char*>(row_bytes.data()), row_bytes.size());
    }
}

int main(int argc, char* argv[]) {
    int N = 16000;
    if (argc > 1) {
        N = std::atoi(argv[1]);
    }
    
    mandelbrot(N, "mandelbrot.pbm");
    return 0;
}